storage/engine/mvcc.go

```diff
@@ -745,7 +745,9 @@ func MVCCIncrement(engine Engine, ms *proto.MVCCStats, key proto.Key, timestamp
 		}
 	}
 
-	int64Val := value.GetInteger()
```

Comment:
could also keep using GetInteger, return its error and remove the check above? Should be cleaner (if the error returned by GetInteger isn't cryptic: `util.Errorf("key %q does not contain an integer value")` comes to mind.